cpp/cmake_modules/FindGandiva.cmake

```diff
@@ -43,7 +43,7 @@ else()
 
   set(GANDIVA_SEARCH_HEADER_PATHS ${GANDIVA_HOME}/include)
 
-  set(GANDIVA_SEARCH_LIB_PATH "${GANDIVA_HOME}")
+  set(GANDIVA_SEARCH_LIB_PATH "${GANDIVA_HOME}/lib")
```

Comment:
We should not hardcode `lib` here as on some systems like Debian this could be in a different location that would be picked up by the default `PATH_SUFFIXES`

Reply:
Otherwise [it searches for gandiva lib under $PREFIX](https://travis-ci.org/kszucs/crossbow/builds/511162234#L2470) and cannot find it. 
Submitted the linux builds to check whether they pass [kszucs/crossbow/build-537](https://github.com/kszucs/crossbow/branches/all?utf8=%E2%9C%93&query=build-537)

Reply:
Additionally see my answer to Wes' question below.